### PR TITLE
Refactor usa-card into component

### PR DESCRIPTION
**Pull request by lcummings12**
## Summary
To reduce duplication code, the`usa-card` partial was enhanced to replace instances of code that could be expressed in terms of `usa-card`.  The `landings/list.html`,  `communities/list.html`  and `guides/list.html` templates all contain inline markup that renders direct variants of `usa-cards` without actually calling the partial. These changes hope to simplify the code by including more potential variation directly in `usa-card`.

## Preview

[Link to Preview]()

## Solution

In the `usa-card` partial, two new parameters were introduced, `site` and `kicker` These parameters allows for further customizability with usages of the partial.  The kicker is part of the original guides cards, and the site parameter is used with `.site.Pages` for filtering communities. 
```
@param {string} [kicker] - Optional kicker text to be displayed above the title
@param {object} [site] - The Hugo site object for context
```
The kicker is part of the original configuration in the guide template. The parameter is needed to duplicate that layout. It is conditionally rendered. 
```
{{ if $kicker }}
  <span class="dg-guides-card__kicker margin-bottom-2">{{- $kicker -}}</span>
{{ end }}
```

The format of the images passed through frontmatter varies, so a check was introduced to determine the format. An image format having a . is a file, so the `$image_path` is set directly to the files name.  If `$image` is not a file, it is a UID, and needs to be passed to `img-flexible.html` to retrieve the image corresponding to the UID. 
```
{{ if (strings.Contains $image ".") }}
  {{ $image_path := (printf "%s/%s" "img" $image | relURL) }}
  <img src="{{- $image_path -}}" alt="{{- $alt -}}" />
{{ else }}
  {{- partial "core/img-flexible.html" (dict "Site" $context "src" $image "small" "false") -}}
{{ end }}
```
Relevant template changes include refactoring to ensure compatibility with the new `usa-card` setup. 

## How To Test
Since the changes focus on code consolidation, a correct implementation should be identical to the production site. 
| Description              | Original URL                                     | Updated URL                                                                                                         |
|--------------------------|--------------------------------------------------|---------------------------------------------------------------------------------------------------------------------|
| Home                     | [Original Home](https://digital.gov/)            | [Updated Home](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/refactor-usa/)              |
| Guides                   | [Original Guides](https://digital.gov/guides/)   | [Updated Guides](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/refactor-usa/guides/)      |
| Guide Landing            | [Original Guide Landing](https://digital.gov/guides/hcd/) | [Updated Guide Landing](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/refactor-usa/guides/hcd/) |
| Communities              | [Original Communities](https://digital.gov/communities/) | [Updated Communities](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/refactor-usa/communities/) |

### CODE REVIEW

**Comment by RileySeaburg** [comment]
@lcummings12 Let's fix these merge issues!

**Comment by lcummings12** [comment]
@RileySeaburg just made a commit that resolves the conflict

**Review by nick-mon1 — CHANGES_REQUESTED** [review]
### Feedback

@lcummings12 I've made some improvements and fixes.

File: themes/digital.gov/layouts/partials/core/featured-communities.html [hunk 1]
```diff
@@ -51,9 +51,12 @@ <h1>{{- .Title | markdownify -}}</h1>
         "image" .Params.dg_logo
         "alt" (printf "%s%s" .Params.dg_shortname " logo")
         "layout" $layout
+        "kicker" ""
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/partials/core/featured-communities.html:54]: ### Suggestion

If we don't use the kicker in this partial call, we can remove this line.

File: themes/digital.gov/layouts/landings/list.html [hunk 2]
```diff
@@ -15,38 +15,25 @@ <h1>{{- .Params.title -}}</h1>
...
-                  >Explore {{ .Title | markdownify | lower }}</a
-                >
-              </div>
-            </div>
+            {{- $image := .Params.image -}}
+            {{- $layout := "no-image" -}}
+            {{- if $image -}}
+              {{- $layout = "has-image" -}}
+            {{- end -}}
+            {{- $cardData := dict "header" .Title "body" .Params.summary "link" (.Permalink | absURL) "link_title" (print "Explore " (.Title | markdownify | lower)) "layout" $layout "site" .Site -}}
```
Comment (nick-mon1) [inline themes/digital.gov/layouts/landings/list.html:25]: ### Fix

Styles are broken for this button.

Screenshot 2024-08-05 at 3 38 12 PM

File: themes/digital.gov/layouts/partials/core/usa-card.html [hunk 3]
```diff
@@ -29,21 +33,32 @@
...
   {{ end }}
+
   {{ if $has_image }}
     {{ if not (eq $image "") }}
-      <div class="usa-card__media">
+      <div class="usa-card__media usa-card__media--indent">
         <div class="usa-card__img">
-          {{ $image_path := (printf "%s/%s" "img" $image | relURL) }}
-          <img src="{{- $image_path -}}" alt="{{- $alt -}}" />
+          {{ if (strings.Contains $image ".") }}
```
[the remark this answers is not in the document]
Comment (lcummings12) [reply themes/digital.gov/layouts/partials/core/usa-card.html:51]: It checks if $image contains a ".", meaning it is stored as a file and is not a UID, and can be directly assigned without first passing through core/img-flexible.html

File: themes/digital.gov/layouts/landings/list.html (hunk 2 above)
Comment (lcummings12) [reply themes/digital.gov/layouts/landings/list.html:25]: this is due to the class originally from the guides page applying universally in usa-card. Im introducing a new optional parameter that adds usa-button--outline if needed.

**Comment by RileySeaburg** [comment]
@lcummings12  did you address each one of the items @nick-mon1  asked for, if so can you click them as addressed?

**Comment by lcummings12** [comment]
@RileySeaburg I did, yes. Most recent commit should have them all

**Review by nick-mon1 — CHANGES_REQUESTED** [review]

**Review by nick-mon1 — APPROVED** [review]
Thanks for tackling this tough challenge @lcummings12, really impressed with the refactoring work. I made some minor adjustments and approved.

**Review by RileySeaburg — APPROVED** [review]
Great job, nice touch introducing new parameters as well.

**Pull request merged**